cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -103,26 +138,44 @@ public InMemoryConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.
    */
   @Override
   public ListenableFuture<ConfigResponse> config() {
-    SettableFuture<ConfigResponse> result = SettableFuture.create();
 
-    try {
-      Object appMain = artifactClassLoader.loadClass(appClassName).newInstance();
-      if (!(appMain instanceof Application)) {
-        throw new IllegalStateException(String.format("Application main class is of invalid type: %s",
-          appMain.getClass().getName()));
+    // Create the classloader
+    ClassLoader tempClassLoader = artifactClassLoader;
+    if (tempClassLoader == null) {
+      EntityImpersonator classLoaderImpersonator = new EntityImpersonator(artifactId.toEntityId(), impersonator);
+      try {
+        tempClassLoader = artifactRepository.createArtifactClassLoader(artifactLocation, classLoaderImpersonator);
+      } catch (Exception e) {
+        return Futures.immediateFailedFuture(e);
       }
+    }
+
+    try (CloseableClassLoader classLoader = (CloseableClassLoader) tempClassLoader) {
```

Comment:
If the artifactclassLoader is passed to this class, we shouldn't close it here.

Reply:
done